### PR TITLE
backend/fix/#730 filtered waypoints above a particular accuracy

### PR DESCRIPTION
**Pull request by soumyajit2304**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Use driver location update accuracy to filter waypoints

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall [hunk 1]
```diff
@@ -123,4 +123,5 @@ in  { esqDBCfg
     , snapToRoadSnippetThreshold = +300
     , minTripDistanceForReferralCfg = Some +1000
     , maxShards = +5
+    , minAccuracy = +50
```
Comment (hkmangla) [inline Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall:126]: add config inside `TransporterConfig`

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall (hunk 1 above)
Comment (soumyajit2304) [reply Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall]: Done

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/Merchant/TransporterConfig.hs [hunk 2]
```diff
@@ -56,7 +56,7 @@ mkPersist
       actualRideDistanceDiffThreshold Centesimal
       upwardsRecomputeBuffer Centesimal
       approxRideDistanceDiffThreshold Centesimal
-
+      minAccuracy Int
```
Comment (piyushKumar-1) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/Merchant/TransporterConfig.hs:59]: why not keep it as `Double` here itself ?

Comment (soumyajit2304) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/Merchant/TransporterConfig.hs]: Done

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs [hunk 3]
```diff
@@ -137,10 +138,12 @@ updateLocationHandler UpdateLocationHandle {..} waypoints = withLogTag "driverLo
     driverInfo <- DInfo.findById (cast driver.id) >>= fromMaybeM (PersonNotFound driver.id.getId)
     logInfo $ "got location updates: " <> getId driver.id <> " " <> encodeToText waypoints
     checkLocationUpdatesRateLimit driver.id
+    thresholdConfig <- QTConf.findByMerchantId driver.merchantId >>= fromMaybeM (InternalError "TransportConfigNotFound")
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Location/UpdateLocation.hs:141]: `fromMaybeM (TransporterConfigNotFound merchantId.getId)`

**Review by roman-bodavskiy — APPROVED** [review]

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**